pymatgen/io/vasp/sets.py

```diff
@@ -1044,6 +1044,185 @@ def __init__(self, structure, potim=0.015, **kwargs):
         self.config_dict["INCAR"].pop("NPAR", None)
 
 
+class MVLGWscDFTSet(DictSet):
```

Comment:
It does not seem like you need four input sets. All the input sets are very similar, and the only difference is the incar default settings. I suggest that you just have either one or two input sets, and one of them can have a "mode", which can stipulate whether it is a SC, Diag, G0W0 or BSE calculation, and then set the incar_settings accordingly.